solidity/contracts/hooks/OPStackHook.sol

```diff
@@ -76,14 +76,11 @@ contract OPStackHook is AbstractMessageIdAuthHook {
         internal
         override
     {
+        uint256 _msgValue = metadata.msgValue(0);
         require(
-            metadata.msgValue() < 2**255,
+            _msgValue < 2**255,
             "OPStackHook: msgValue must less than 2 ** 255"
         );
-        l1Messenger.sendMessage{value: metadata.msgValue()}(
-            ism,
-            payload,
-            GAS_LIMIT
-        );
+        l1Messenger.sendMessage{value: _msgValue}(ism, payload, GAS_LIMIT);
```

Comment:
I think actually keeping it as calldata slicing instead of loading it into memory should be slightly cheaper